src/panels/config/automation/dialog-new-automation.ts

```diff
@@ -33,9 +34,13 @@ class DialogNewAutomation extends LitElement {
     this._opened = true;
   }
 
-  public closeDialog(): void {
+  public async closeDialog(): Promise<void> {
+    if (!this._opened) {
```

Comment:
Without this guard, it would fire twice because of the `@closed` event listener on `<ha-dialog>`